app/templates/components/project-card.hbs

```diff
@@ -13,6 +13,29 @@
   <p class="description">
     {{project.description}}
   </p>
+  <p class="project-card__actions">
+    {{#if project.shouldLinkExternally}}
+      <a class="button default small external-link" href="{{project.website}}">Join project<span></span></a>
+    {{else}}
+      {{#if session.isAuthenticated}}
+        {{#if (and currentProjectMembership currentProjectMembership.isLoaded)}}
+          {{#if (eq currentProjectMembership.role 'pending')}}
+            <button class="clear small" disabled>Request sent</button>
+          {{else}}
+            <button class="clear small" disabled>Member</button>
+          {{/if}}
+        {{/if}}
+
+        {{#unless currentProjectMembership}}
+          <button class="button default small" {{action "toggleJoinProjectModal"}}>Join project</button>
+        {{/unless}}
+      {{else}}
+        {{link-to "Join project" "signup" class="button default small"}}
+      {{/if}}
+    {{/if}}
```

Comment:
I've simplified this some by moving the modal and the button that opens it into the same component. My suggestion for further simplification would be to do some extraction, but probably in a separate PR.

The two base cases are:

- There is a membership
- There is no membership

For the first one, show a `membership-status` component with 3 states

- loading
- request sent (role eq pending)
- member (role eq something else)

For the second one, we just use the current component.

```hbs
{{#if session.isAuthenticated}}
  {{#if currentMembership}}
    {{membership-status membership=currentMembership}}
  {{else}}
    {{project-join-modal onJoin=(action onJoin project) project=project skills=projectSkills}}
  {{/if}}
{{else}}
  {{link-to "Join project" "signup" class="button default small"}}
{{/if}}
```